### PR TITLE
Config validation

### PR DESCRIPTION
**Pull request by ftassi**
This adds some validation to the Config object inner data. Have a look at `\Idephix\ConfigTest` in particular for the messages returned to the user.

This combined to a more complete and expressive `idxrc.php` (with all the defaults commented) should be enough to ease ideaphix setup

### CODE REVIEW

**Comment by micheleorselli** [comment]
Some files are checked in the bin directory, like 'box', 'phpunit', 'php-cs-fixer'. Was it intended? Afaik they should be excluded since they are managed by composer

**Comment by ftassi** [comment]
See 102ae98 I removed some deps from composer and switched to phar to avoid some dependencies issues. The other solution would have been to define the exact version in the composer.json file but since they are all dev deps I preferred the phar

File: docs/Cookbook/Deploy/idxrc.php [hunk 1]
```diff
@@ -19,9 +24,9 @@
 
 return \Idephix\Config::fromArray(
     array(
-        \Idephix\Config::ENVS => $environments,
-        \Idephix\Config::SSHCLIENT => new \Idephix\SSH\SshClient(new \Idephix\SSH\CLISshProxy()),
-        \Idephix\Config::EXTENSIONS => array(
+        'envs' => $environments,
+        'ssh_client' => new \Idephix\SSH\SshClient(new \Idephix\SSH\CLISshProxy()),
```
Comment (micheleorselli) [inline docs/Cookbook/Deploy/idxrc.php]: is `CLISshProxy` an optional dependency? Asking since in the doc seems it's not needed

Comment (ftassi) [inline docs/Cookbook/Deploy/idxrc.php]: Yes, it is optional indeed: SshClient will create a CliSshProxy instance in null is passed

Comment (micheleorselli) [inline docs/Cookbook/Deploy/idxrc.php]: Maybe we can remove it for clarity

**Pull request merged**